### PR TITLE
IntSights plugin

### PR DESCRIPTION
**Pull request by r7-mgorny**
## Proposed Changes
* Get Indicator by Value
* Enrich Indicator
* Rescan Indicator
* Get Indicator Scan Status
* Get Alerts
* Get Complete Alert by ID
* Takedown Request
* Add Manual Alert

### Description

Describe the proposed changes:

 - Init new plugin: IntSights

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Run

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/https://example.com Step name: add_manual_alert\n",
    "meta": {},
    "output": {
      "alert_id": "6156586e8eadf90008176450"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/intsights:1.0.0 --debug run < tests/add_manual_alert.json
</summary>
</details>

<details>

```
{
  'OriginalValue': 'example.com',
  'Status': 'Done',
  'Data': {
    'Value': 'example.com',
    'Type': 'Domains',
    'IsKnownIoc': False,
    'Whitelisted': True,
    'Tags': [],
    'Sources': [],
    'Severity': {
      'Value': 'Low',
      'Score': 0
    },
    'SystemTags': [],
    'RelatedMalwares': [],
    'RelatedThreatActors': [],
    'RelatedCampaigns': [],
    'DnsRecords': [
      {
        'Value': 'ns1.example.com.',
        'Type': 'NS',
        'FirstResolved': None,
        'LastResolved': None,
        'Count': 135
      },
      {
        'Value': 'ns2.example.com.',
        'Type': 'NS',
        'FirstResolved': None,
        'LastResolved': None,
        'Count': 135
      },
      {
        'Value': 'ns1.example.com.',
        'Type': 'NS',
        'FirstResolved': None,
        'LastResolved': None,
        'Count': 3072
      }
    ]
  }
}

```

<summary>
docker run --rm -i rapid7/intsights:1.0.0 --debug run < tests/enrich_indicator.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/https://example.com Step name: get_alerts\n",
    "meta": {},
    "output": {
      "alert_ids": [
        "6155f7b7c6e9d40008b4bb0d",
        "6155f801186a050007745d29",
        "6156114d320a3d0007ca4fd3",
        "615611781e91570008501448",
        "61561181186a05000774ee00",
        "61561187186a05000774ee19",
        "6156118e186a05000774ee46",
        "6156119f628ea5000763533c",
        "615617411e9157000850352f",
        "615645bac6e9d40008b69a07",
        "61565741628ea5000765198f",
        "615657d38eadf900081760f3",
        "6156586e8eadf90008176450"
      ]
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/intsights:1.0.0 --debug run < tests/get_alerts.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/https://example.com Step name: get_complete_alert_by_id\nresponse: {'_id': '6156118e186a05000774ee46', 'Details': {'Source': {'Type': 'Application Store', 'NetworkType': 'ClearWeb', 'URL': 'https://example.com'}, 'Images': [], 'Title': 'Alerttest3', 'Type': 'Phishing', 'SubType': 'SuspiciousEmailAddress', 'Severity': 'High', 'Tags': [], 'Description': 'APIDescription'}, 'Assignees': [], 'FoundDate': 'https://example.com', 'Assets': [], 'TakedownStatus': 'NotSent', 'IsFlagged': False, 'UpdateDate': 'https://example.com', 'RelatedIocs': [], 'RelatedThreatIDs': [], 'Closed': {'IsClosed': False}}\n",
    "meta": {},
    "output": {
      "assets": [],
      "assignees": [],
      "details": {
        "Description": "APIDescription",
        "Images": [],
        "Severity": "High",
        "Source": {
          "NetworkType": "ClearWeb",
          "Type": "Application Store",
          "URL": "https://example.com"
        },
        "SubType": "SuspiciousEmailAddress",
        "Tags": [],
        "Title": "Alerttest3",
        "Type": "Phishing"
      },
      "found_date": "https://example.com",
      "id": "6156118e186a05000774ee46",
      "is_closed": false,
      "is_flagged": false,
      "takedown_status": "NotSent",
      "update_date": "https://example.com"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/intsights:1.0.0 --debug run < tests/get_complete_alert_by_id.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/https://example.com Step name: get_indicator_by_value\nresponse: {'Value': 'https://example.com', 'Type': 'Domains', 'Score': 10, 'Severity': 'Low', 'Whitelist': True, 'FirstSeen': 'https://example.com', 'LastSeen': 'https://example.com', 'LastUpdate': 'https://example.com', 'Sources': [{'ConfidenceLevel': 2, 'Name': 'Cyber Threat Alliance'}], 'SystemTags': [], 'RelatedMalware': [], 'RelatedCampaigns': [], 'RelatedThreatActors': [], 'Tags': []}\n",
    "meta": {},
    "output": {
      "first_seen": "https://example.com",
      "last_seen": "https://example.com",
      "last_update": "https://example.com",
      "related_campaigns": [],
      "related_malware": [],
      "related_threat_actors": [],
      "score": 10,
      "severity": "Low",
      "sources": [
        {
          "ConfidenceLevel": 2,
          "Name": "Cyber Threat Alliance"
        }
      ],
      "system_tags": [],
      "tags": [],
      "type": "Domains",
      "value": "https://example.com",
      "whitelist": true
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/intsights:1.0.0 --debug run < tests/get_indicator_by_value.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/https://example.com Step name: get_indicator_scan_status\nresponse: {'TaskId': '61563eb2118b97e8e388e9db', 'Status': 'Done'}\n",
    "meta": {},
    "output": {
      "status": "Done",
      "task_id": "61563eb2118b97e8e388e9db"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/intsights:1.0.0 --debug run < tests/get_indicator_scan_status.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/https://example.com Step name: rescan_indicator\nresponse: {'TaskId': '615658811baf672bdaeb8e5c', 'Status': 'Queued'}\n",
    "meta": {},
    "output": {
      "status": "Queued",
      "task_id": "615658811baf672bdaeb8e5c"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/intsights:1.0.0 --debug run < tests/rescan_indicator.json
</summary>
</details>

### Test

Autogenerate with:
<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/https://example.com Step name: get_alerts\n",
    "meta": {},
    "output": {
      "success": true
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/intsights:1.0.0 --debug test < tests/get_alerts.json
</summary>
</details>

### CODE REVIEW

File: plugins/intsights/plugin.spec.yaml [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+- threatintelligence
+hub_tags:
+  use_cases: [data_utility, threat_detection_and_response]
+  keywords: [threat_intelligence, phishing, remediation, block, malware_analysis]
+  features: []
+types:
+  source:
+    Name:
+      title: Name
+      descirption: Name
```
Comment (jrose-r7) [inline plugins/intsights/plugin.spec.yaml:28]:
```suggestion
      description: Name
```

**Review by jrose-r7 — CHANGES_REQUESTED** [review]

**Comment by jrose-r7** [comment]
Looks like there's a few failing validators because of the help.md
Screen Shot 2021-10-05 at 10 59 42 AM
It might prefer using example.com instead of example.com instead of https://example.com

File: plugins/intsights/icon_intsights/actions/get_indicator_by_value/action.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            {
+                Output.VALUE: response.get("Value"),
+                Output.TYPE: response.get("Type"),
+                Output.SEVERITY: response.get("Severity"),
+                Output.SCORE: response.get("Score", 0),
+                Output.WHITELIST: response.get("Whitelist", False),
+                Output.FIRST_SEEN: response.get("FirstSeen"),
+                Output.LAST_SEEN: response.get("LastSeen"),
+                Output.LAST_UPDATE: response.get("LastUpdate"),
+                Output.GEO_LOCATION: response.get("Geolocation"),
```
Comment (mrinehart-r7) [inline plugins/intsights/icon_intsights/actions/get_indicator_by_value/action.py:29]: Do any of these need to be defaulted?

File: plugins/intsights/icon_intsights/actions/takedown_request/action.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        super(self.__class__, self).__init__(
+            name="takedown_request",
+            description=Component.DESCRIPTION,
+            input=TakedownRequestInput(),
+            output=TakedownRequestOutput(),
+        )
+
+    def run(self, params={}):
+        self.connection.client.takedown_request(params.get(Input.ALERT_ID), params.get(Input.TARGET))
+        return clean({Output.STATUS: True})
```
Comment (mrinehart-r7) [inline plugins/intsights/icon_intsights/actions/takedown_request/action.py:19]:
```suggestion
        return {Output.STATUS: True}
```
No need for clean here if this is always True.

File: plugins/intsights/icon_intsights/connection/connection.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import insightconnect_plugin_runtime
+from typing import Optional
+from .schema import ConnectionSchema, Input
+
+# Custom imports below
+from icon_intsights.util.api import IntSightAPI
+from insightconnect_plugin_runtime.exceptions import PluginException, ConnectionTestException
```
Comment (mrinehart-r7) [inline plugins/intsights/icon_intsights/connection/connection.py:7]:
```suggestion
from .schema import ConnectionSchema, Input

# Custom imports below
from typing import Optional
from icon_intsights.util.api import IntSightAPI
from insightconnect_plugin_runtime.exceptions import PluginException, ConnectionTestException
```

File: plugins/intsights/icon_intsights/util/api.py [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+                        "URL": self.source_url,
+                        "Date": self.source_date,
+                    },
+                    "Images": images,
+                },
+            }
+        )
+
+
+class IntSightAPI:
```
Comment (mrinehart-r7) [inline plugins/intsights/icon_intsights/util/api.py:101]:
```suggestion
class IntSightsAPI:
```

File: plugins/intsights/icon_intsights/util/api.py [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+        try:
+            response = self.make_request(method=method, path=path, json_data=json_data, params=params)
+            if response.status_code == 204:
+                return {}
+
+            json_response = response.json()
+
+            if json_response.get("Status") == "Invalid":
+                raise PluginException(
+                    cause="There is an error in response.", assistance=f"{json_response.get('FailedReason')}."
```
Comment (mrinehart-r7) [inline plugins/intsights/icon_intsights/util/api.py:153]:
```suggestion
                    cause="IntSights returned an error response: ", assistance=f"{json_response.get('FailedReason')}."
```

**Review by mrinehart-r7 — DISMISSED** [review]

**Review by mberezin-r7 — APPROVED** [review]
Aside from the potential infinite loop PJ commented on, this looks good to me.

**Review by mberezin-r7 — APPROVED** [review]

**Review by jrose-r7 — DISMISSED** [review]

File: plugins/rapid7_intsights/help.md [hunk 7]
```diff
@@ -72,10 +72,7 @@ Example input:
 {
   "description": "Suspicious addresses",
   "found_date": "2020-01-01",
-  "images": [{
-    "Type": "jpeg",
-    "Data": "UmFwaWQ3IEluc2lnaHRDb25uZWN0Cg=="
-  }],
+  "images": "[{\"Type\": \"jpeg\",\"Data\": \"UmFwaWQ3IEluc2lnaHRDb25uZWN0Cg==\"}]",
```
Comment (mrinehart-r7) [inline plugins/rapid7_intsights/help.md:75]:
```suggestion
  "images": [{
    "Type": "jpeg",
    "Data": "UmFwaWQ3IEluc2lnaHRDb25uZWN0Cg=="
  }],
```

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

**Review by mrinehart-r7 — APPROVED** [review]

**Review by mberezin-r7 — APPROVED** [review]

**Comment by r7-mgorny** [comment]
## Assessment
### Run

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Rapid7 https://example.com Step name: enrich_indicator\n",
    "meta": {},
    "output": {
      "data": {
        "DnsRecords": [
          {
            "Count": 16523,
            "FirstResolved": "https://example.com",
            "LastResolved": "https://example.com",
            "Type": "NS",
            "Value": "https://example.com"
          },
          {
            "Count": 16523,
            "FirstResolved": "https://example.com",
            "LastResolved": "https://example.com",
            "Type": "NS",
            "Value": "https://example.com"
          },
          {
            "Count": 646,
            "FirstResolved": "https://example.com",
            "LastResolved": "https://example.com",
            "Type": "NS",
            "Value": "https://example.com"
          }
        ],
        "IsKnownIoc": false,
        "RelatedCampaigns": [],
        "RelatedHashes": {
          "communicating": [
            "22n708a98wxww6p76lif8r7rn1a7422012l579m82i222e68957u4603510y4228"
          ],
          "downloaded": [
            "22n708a98wxww6p76lif8r7rn1a7422012l579m82i222e68957u4603510y4228"
          ],
          "referencing": []
        },
        "RelatedMalwares": [],
        "RelatedThreatActors": [],
        "Resolutions": [
          {
            "ASN": 16276,
            "FirstResolved": "https://example.com",
            "LastResolved": "https://example.com",
            "Location": "DE",
            "Operator": "OVH SAS",
            "ReportingSources": [
              "Farsight",
              "VirusTotal"
            ],
            "ResolvedIpAddress": "https://example.com"
          },
          {
            "ASN": 16276,
            "FirstResolved": "https://example.com",
            "LastResolved": "https://example.com",
            "Location": "FR",
            "Operator": "OVH SAS",
            "ReportingSources": [
              "Farsight",
              "VirusTotal"
            ],
            "ResolvedIpAddress": "https://example.com"
          }
        ],
        "Severity": {
          "Score": 20,
          "Value": "Low"
        },
        "Sources": [],
        "Subdomains": [
          "https://example.com",
          "metami",
          "test829"
        ],
        "SystemTags": [],
        "Tags": [],
        "Type": "Domains",
        "Value": "https://example.com",
        "Whitelisted": false,
        "Whois": {
          "Current": {
            "RegistrantDetails": [],
            "RegistrationDetails": {
              "CreatedDate": "https://example.com",
              "ExpiresDate": "https://example.com",
              "NameServers": [
                "https://example.com",
                "https://example.com"
              ],
              "Statuses": [],
              "UpdatedDate": "https://example.com"
            }
          },
          "History": [
            {
              "RegistrantDetails": [],
              "RegistrationDetails": {
                "CreatedDate": "https://example.com",
                "ExpiresDate": "https://example.com",
                "NameServers": [
                  "https://example.com",
                  "https://example.com"
                ],
                "Statuses": [],
                "UpdatedDate": "https://example.com"
              }
            }
          ]
        }
      },
      "original_value": "https://example.com",
      "status": "Done"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/rapid7_intsights:1.0.0 --debug run < tests/enrich_indicator.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Rapid7 https://example.com Step name: get_alerts\n",
    "meta": {},
    "output": {
      "alert_ids": [
        "3396n3b4z9s2b77964e1fp4e",
        "3599v148818t995862708l85",
        "4618302j551a6s3749un5gq5"
      ]
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/rapid7_intsights:1.0.0 --debug run < tests/get_alerts.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Rapid7 https://example.com Step name: get_complete_alert_by_id\n",
    "meta": {},
    "output": {
      "assets": [],
      "assignees": [],
      "details": {
        "Description": "Alert description",
        "Images": [],
        "Severity": "High",
        "Source": {
          "NetworkType": "ClearWeb",
          "Type": "Application Store",
          "URL": "https://example.com"
        },
        "SubType": "SuspiciousEmailAddress",
        "Tags": [],
        "Title": "Alert test 4",
        "Type": "Phishing"
      },
      "found_date": "https://example.com",
      "id": "29899f708d7j6a1628ck5p50",
      "is_closed": false,
      "is_flagged": false,
      "takedown_status": "NotSent",
      "update_date": "https://example.com"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/rapid7_intsights:1.0.0 --debug run < tests/get_complete_alert_by_id.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Rapid7 https://example.com Step name: get_indicator_by_value\n",
    "meta": {},
    "output": {
      "first_seen": "https://example.com",
      "last_seen": "https://example.com",
      "last_update": "https://example.com",
      "related_campaigns": [],
      "related_malware": [],
      "related_threat_actors": [],
      "score": 10,
      "severity": "Low",
      "sources": [
        {
          "ConfidenceLevel": 2,
          "Name": "Cyber Threat Alliance"
        }
      ],
      "system_tags": [],
      "tags": [],
      "type": "Domains",
      "value": "https://example.com",
      "whitelist": true
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/rapid7_intsights:1.0.0 --debug run < tests/get_indicator_by_value.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Rapid7 https://example.com Step name: rescan_indicator\n",
    "meta": {},
    "output": {
      "status": "Queued",
      "task_id": "4328n08s02j33e0f27oe1h8p"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/rapid7_intsights:1.0.0 --debug run < tests/rescan_indicator.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Rapid7 https://example.com Step name: get_indicator_scan_status\n",
    "meta": {},
    "output": {
      "status": "Done",
      "task_id": "55638xdla173d41713aabh14"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/rapid7_intsights:1.0.0 --debug run < tests/get_indicator_scan_status.json
</summary>
</details>

<details>

```
{
  "body": {
    "error": "An error occurred during plugin execution!\n\nInvalid API key provided. Verify your API key configured in your connection is correct. Response was: RemediationLimitExceeded",
    "log": "Connect: Connecting...\nrapid7/Rapid7 https://example.com Step name: takedown_request\nAn error occurred during plugin execution!\n\nInvalid API key provided. Verify your API key configured in your connection is correct. Response was: RemediationLimitExceeded\nTraceback (most recent call last):\n  File \"/usr/local/lib/https://example.com\", line 326, in handle_step\n    output = https://example.com\n  File \"/usr/local/lib/https://example.com\", line 476, in start_step\n    output = func(params)\n  File \"/usr/local/lib/https://example.com\", line 17, in run\n    https://example.com, https://example.com\n  File \"/usr/local/lib/https://example.com\", line 131, in takedown_request\n    return https://example.com\n  File \"/usr/local/lib/https://example.com\", line 146, in make_json_request\n    response = https://example.com, path=path, json_data=json_data, params=params)\n  File \"/usr/local/lib/https://example.com\", line 176, in make_request\n    raise PluginException(https://example.com, https://example.com\https://example.com An error occurred during plugin execution!\n\nInvalid API key provided. Verify your API key configured in your connection is correct. Response was: RemediationLimitExceeded\n",
    "meta": {},
    "status": "error"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/rapid7_intsights:1.0.0 --debug run < tests/takedown_request.json
</summary>
</details>

### Test

Autogenerate with:
<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/Rapid7 https://example.com Step name: add_manual_alert\n",
    "meta": {},
    "output": {
      "success": true
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/rapid7_intsights:1.0.0 --debug test < tests/add_manual_alert.json
</summary>
</details>

### Validate

Autogenerate with:
<details>

```
[*] Use ``make menu`` for available targets
[*] Including available Makefiles: ../../tools/Makefiles/Colors.mk ../../tools/Makefiles/Helpers.mk
--
[*] Running validators
DEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621
[*] Validating plugin at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator CloudReadyConnectionCredentialTokenValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator VersionPinValidator
[*] Executing validator EncodingValidator
[*] Executing validator ExampleInputValidator
[*] Executing validator CloudReadyValidator
[*] Executing validator SupportedVersionValidator
[*] Executing validator UnapprovedKeywordsValidator
[*] Executing validator HelpExampleValidator
[*] Executing validator Version Increment Validator
[*] Plugin failed validation! The following validation errors occurred:

Validator "HelpExampleValidator" failed! 
	Cause: Invalid JSON example identified in Example input of action titled Add Manual Alert. Please correct the JSON example in help.md.

Validator "Version Increment Validator" failed! 
	Cause: Remote origin/master not found.'master' branch name changed, update validator


----
[*] Total time elapsed: 3093.517ms

```

<summary>
make validate
</summary>
</details>

<details>

```
[*] Validating plugin with all validators at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator CloudReadyConnectionCredentialTokenValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator VersionPinValidator
[*] Executing validator EncodingValidator
[*] Executing validator ExampleInputValidator
[*] Executing validator CloudReadyValidator
[*] Executing validator SupportedVersionValidator
[*] Executing validator UnapprovedKeywordsValidator
[*] Executing validator HelpExampleValidator
[*] Executing validator Version Increment Validator
[*] Executing validator ExceptionValidator
[*] Executing validator CredentialsValidator
[*] Executing validator PasswordValidator
[*] Executing validator PrintValidator
[*] Executing validator ConfidentialValidator
[*] Executing validator DockerValidator
[*] Executing validator URLValidator
[*] Plugin failed validation! The following validation errors occurred:

Validator "HelpExampleValidator" failed! 
	Cause: Invalid JSON example identified in Example input of action titled Add Manual Alert. Please correct the JSON example in help.md.

Validator "Version Increment Validator" failed! 
	Cause: Remote origin/master not found.'master' branch name changed, update validator


----
[*] Total time elapsed: 22218.535ms

```

<summary>
icon-validate --all .
</summary>
</details>

**Review by mrinehart-r7 — APPROVED** [review]

**Review by mberezin-r7 — APPROVED** [review]

**Pull request merged**